.github/actions/secure-download-artifact/action.yml

```diff
@@ -24,23 +24,30 @@ runs:
       run: |
         set -euo pipefail
 
-        # Check that the name of the path does not clash with the
-        # directory where the builder code is stored.
+        # Ensure no directory traversal.
+        # NOTE: the actions/download-artifact Action only creates files
+        # in the workspace directory, but this may change in the future.
+        resolved_dir=$(dirname $(readlink -m "$UNTRUSTED_PATH"))
+        if [[ "$resolved_dir" != "$PWD" ]]; then
```

Comment:
Why would calling readlink on PWD be required?

Reply:
> Why would calling readlink on PWD be required?

I just wasn't sure if PWD is always the canonical path. You'd want it to be a canonical path for comparison.

> Would your proposed changed catch a resolved_dirs="/etc/password"? Did you mean !=?

ah yes. my bad. It should be a `!=`.

```suggestion
        resolved_dir=$(readlink -m "$UNTRUSTED_PATH")
        wd=$(readlink -m "$PWD")
        if [[ "$resolved_dir" != "$wd"/* ]]; then
```

Reply:
PWD is always canonical, but let's resolve anyway, does not hurt.

Reply:
> PWD is always canonical, but let's resolve anyway, does not hurt.

Yeah. Env vars are fairly easy to manipulate so I think it's a good idea just in case.

Reply:
auto-merge was on, my bad. I need to fix this.